sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/schema/BeamSqlRecord.java

```diff
@@ -17,24 +17,25 @@
  */
 package org.apache.beam.sdk.extensions.sql.schema;
 
-import com.google.auto.value.AutoValue;
-import java.io.Serializable;
 import java.util.List;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.PCollection;
 
 /**
- * Field type information in {@link BeamSqlRow}.
- *
+ * {@link BeamSqlRecord} represents one row element in a {@link PCollection},
+ * with type provider {@link BeamSqlRecordTypeProvider}.
  */
-@AutoValue
-public abstract class BeamSqlRowType implements Serializable {
-  public abstract List<String> getFieldsName();
-  public abstract List<Integer> getFieldsType();
+public class BeamSqlRecord extends BeamRecord {
```

Comment:
If BeamSqlRecord  is just a helper class, it shouldn't be in the public API. If that means a couple of casts to `BeamSqlRecordTypeProvider` then that's OK. However, the only methods I see on this class are `validateValueType` (rarely used) and `getFieldsType`. 

Of course `getFieldsType` (should this be `getFieldTypes`?) the interesting one. However, nearly every use of this method is to create a new `BeamSqlRecordTypeProvider`. It would actually be better to add methods on `BeamRecordType` that slice/concatenate/etc. to create new `BeamRecordType`s (with overloads in `BeamSqlRecordTypeProvider` to create `BeamSqlRecordType`s). 

Done right, simple SQL statements such as selections, joins, and projections should work on generic BeamRecords. Aggregations and comparisons would still (likely) require knowing the specific types (as SQL Types) and hence an actual `BeamSqlRecordType`, but that's OK as we're already in the domain of throwing errors if it's not the right type (and a generic BeamRecordType schema would always throw a "wrong type" error.) We don't, of course, have to get this working now. 

The naming still seems to be in flux. I'd drop the `Provider` suffix, and get rid of `Row` everywhere. The suite of types would then be.

* `BeamRecord`
* `BeamRecordCoder`
* `BeamSqlRecordType` extends `BeamRecordType`